modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -0,0 +1,190 @@
+import org.wso2.charon3.core.exceptions.InternalErrorException;
+import org.wso2.charon3.core.exceptions.NotFoundException;
+import org.wso2.charon3.core.extensions.UserManager;
+import org.wso2.charon3.core.objects.User;
+import org.wso2.charon3.core.protocol.SCIMResponse;
+import org.wso2.charon3.core.schema.SCIMConstants;
+import org.wso2.charon3.core.schema.SCIMResourceSchemaManager;
+import org.wso2.charon3.core.schema.SCIMResourceTypeSchema;
+import org.wso2.charon3.core.utils.CopyUtil;
+import org.wso2.charon3.core.utils.ResourceManagerUtil;
+
+import java.util.Map;
+
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/**
+ * Test class of UserResourceManager.
+ */
+@PrepareForTest({AbstractResourceManager.class})
+public class UserResourceManagerTest extends PowerMockTestCase {
+
+    private UserResourceManager userResourceManager = new UserResourceManager();
+
+    @Mock
+    private UserManager userManager;
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+
+        return new Object[][]{
+                {id, "userName", null, 200, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetSuccess")
+    public void testGetSuccess(String id, String attributes,
+                               String excludeAttributes, int expectedScimResponseStatus, Object objectUser)
+                               throws BadRequestException, NotFoundException, CharonException {
+
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Users");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getUser(id, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        Assert.assertEquals(obj.getString("id"), id);
+    }
+
+    @DataProvider(name = "dataForGetThrowingExceptions")
+    public Object[][] dataToGetThrowingExceptions() {
```

Comment:
Better to add a comment with what this data holder cotains

Reply:
Fixed with 39f1812bad7f3cc5344e1b3327f8f54e5e995532